src/http/request.cr

```diff
@@ -246,14 +246,40 @@ class HTTP::Request
   end
 
   # Returns request host from headers.
+  @[Deprecated("Use `#hostname` instead.")]
   def host
     host = @headers["Host"]?
     return unless host
     index = host.index(":")
     index ? host[0...index] : host
   end
 
+  # Extracts the hostname from `Host` header.
+  #
+  # Returns `nil` if `Host` header is missing.
+  #
+  # If the `Host` header contains a port number, it is stripped off.
+  def hostname : String?
+    header = @headers["Host"]?
+    return unless header
+
+    host, _, port = header.rpartition(":")
+    if host.nil?
```

Comment:
```suggestion
    if host.empty?
```
`String#rpartition` returns `Tuple(String, String, String)`

Reply:
Oh yeah, that entire branch is unnecessary (and potentially wrong because it doesn't unwrap IP addresses).